### ISSUE TITLE
BUG: Inconsistent behavior in plot_dataframe when using aspect=None with missing_kwds

### ISSUE THREAD

**Issue by kbonney** [geopandas/geopandas#3432]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

#### Code Sample, a copy-pastable example

```python
import geopandas as gpd
import matplotlib.pyplot as plt
from shapely.geometry import Point

data = {
    'geometry': [Point(1, 1), Point(5, 5), Point(5, 10)],
    'value': [10, None, 10]
}

gdf = gpd.GeoDataFrame(data, crs="EPSG:4326")

fig, axes = plt.subplots(2,1)

gdf.plot(column='value', legend=True, aspect=None, ax=axes[0])
print("No missing_kwds, axes[0].get_aspect() = ", axes[0].get_aspect())

gdf.plot(column='value', missing_kwds={"color": "lightgray"}, legend=True, aspect=None, ax=axes[1])
print("With missing_kwds, axes[1].get_aspect() = ", axes[1].get_aspect())

plt.show()

```

#### Problem description

When plotting with `aspect=None` the aspect ratio of the resulting plot changes depending on whether or not a value for `missing_kwds` is passed. When no value for`missing_kwds` is passed, the aspect on the axis is "auto", as expected. However, when a value for `missing_kwds` is passed, the aspect on the axis is approximately 1.

#### Expected Output
The expected output when a value for `missing_kwds` is passed to `plot_dataframe` is an aspect value that is the same as the result when plotting without a passing a value for `missing_kwds`. In this case, the aspect should remain as `auto`.

#### A potential fix
I am able to fix this particular case by changing the line `geopandas/plotting.py:890` from
```
        plot_series(expl_series[nan_idx], ax=ax, **merged_kwds)
```
to
```
        plot_series(expl_series[nan_idx], ax=ax, aspect=aspect, **merged_kwds)
```
The problem is that `plot_series` has a default value for aspect which overrides the argument originally passed by the user. By passing the user's value to this internal call of `plot_series` , the expected output is produced.

If the maintainers agree with this solution to the problem, I am happy to resolve this issue via a PR with the above fix.

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]
executable : /home/klbonne/.local/miniconda3/envs/wntr/bin/python
machine    : Linux-6.3.13-060313-generic-x86_64-with-glibc2.35

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.11.1
GEOS lib   : None
GDAL       : 3.8.5
GDAL data dir: /home/klbonne/.local/miniconda3/envs/wntr/lib/python3.10/site-packages/pyogrio/gdal_data/
PROJ       : 9.2.1
PROJ data dir: /home/klbonne/.local/miniconda3/envs/wntr/lib/python3.10/site-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 1.0.1
numpy      : 1.26.4
pandas     : 2.0.3
pyproj     : 3.6.0
shapely    : 2.0.1
pyogrio    : 0.9.0
geoalchemy2: None
geopy      : None
matplotlib : 3.7.2
mapclassify: None
fiona      : 1.9.6
psycopg    : None
psycopg2   : None
pyarrow    : None

SYSTEM INFO
-----------
python     : 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]
executable : /home/klbonne/.local/miniconda3/envs/wntr/bin/python
machine    : Linux-6.3.13-060313-generic-x86_64-with-glibc2.35

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.11.1
GEOS lib   : None
GDAL       : 3.8.5
GDAL data dir: /home/klbonne/.local/miniconda3/envs/wntr/lib/python3.10/site-packages/pyogrio/gdal_data/
PROJ       : 9.2.1
PROJ data dir: /home/klbonne/.local/miniconda3/envs/wntr/lib/python3.10/site-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 1.0.1
numpy      : 1.26.4
pandas     : 2.0.3
pyproj     : 3.6.0
shapely    : 2.0.1
pyogrio    : 0.9.0
geoalchemy2: None
geopy      : None
matplotlib : 3.7.2
mapclassify: None
fiona      : 1.9.6
psycopg    : None
psycopg2   : None
pyarrow    : None

</details>

**Comment by martinfleis** [comment]
Thanks! Fix is coming in #3438 

> `aspect=aspect`

It has to be `aspect=None`, otherwise with EPSG 4326, we would recompute the aspect.

**Comment by kbonney** [comment]
Thanks, @martinfleis.
